datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1138,7 +1139,7 @@ fn from_substrait_type(dt: &substrait::proto::Type) -> Result<DataType> {
                     from_substrait_type(list.r#type.as_ref().ok_or_else(|| {
                         substrait_datafusion_err!("List type must have inner type")
                     })?)?;
-                let field = Arc::new(Field::new("list_item", inner_type, true));
+                let field = Arc::new(Field::new_list_field(inner_type, true));
```

Comment:
this is a breaking change in the sense that the new field name is just "item" - to align with Arrow default